spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractMessageListenerContainer.java

```diff
@@ -167,6 +168,14 @@ protected AbstractMessageListenerContainer(ConsumerFactory<? super K, ? super V>
 		if (this.containerProperties.getConsumerRebalanceListener() == null) {
 			this.containerProperties.setConsumerRebalanceListener(createSimpleLoggingConsumerRebalanceListener());
 		}
+		final OffsetCommitCallback commitCallback = containerProperties.getCommitCallback();
+		if (commitCallback != null) {
+			this.containerProperties.setCommitCallback(commitCallback);
+		}
+		final OffsetAndMetadataProvider offsetAndMetadataProvider = containerProperties.getOffsetAndMetadataProvider();
+		if (offsetAndMetadataProvider != null) {
+			this.containerProperties.setOffsetAndMetadataProvider(containerProperties.getOffsetAndMetadataProvider());
+		}
```

Comment:
This code is unnecessary; these properties were already copied by `BeanUtils.copyProperties()` above.

Reply:
Useless code has bee removed for both OffsetCommitCallback and OffsetAndMetadataProvider.
Thanks for the review.